### PR TITLE
Kafka: Record the initial high watermark

### PR DESCRIPTION
**Pull request by antiguru**
Signed-off-by: Moritz Hoffmann <mh@materialize.com>

### CODE REVIEW

File: doc/user/content/sql/system-catalog.md [hunk 1]
```diff
@@ -270,6 +270,7 @@ Field           | Type       | Meaning
 `ls_offset`     | [`bigint`] | The partition's last stable offset on the broker.
 `app_offset`    | [`bigint`] | The offset of the last message passed to materialized + 1.
 `consumer_lag`  | [`bigint`] | Difference between the `hi_offset` and `app_offset`.
+`initial_hi_offset`  | [`bigint`] | The first known partition's high watermark offset on the broker.
```
Comment (antifuchs) [inline doc/user/content/sql/system-catalog.md:273]: Hah, the "lo" / "hi" / "slo" / "fas" abbreviation thing is a bit of a pet peeve of mine: can we name it `initial_high_offset`?

File: doc/user/content/release-notes.md [hunk 2]
```diff
@@ -53,6 +53,9 @@ Wrap your release notes at the 80 character mark.
   [postgres sources](/sql/create-source/postgres/#syntax) has been renamed to
   `CONNECTION`.
 
+- Record the initial high watermark offset on the broker for Kafka sources. This
+  enables clients to observe the progress of initial data loading.
```
Comment (antifuchs) [inline doc/user/content/release-notes.md:57]: Make sure to mention the table `mz_kafka_consumer_partitions` here! (:

**Review by antifuchs — APPROVED** [review]
This looks good to me, cc @elindsey (:

File: doc/user/content/sql/system-catalog.md (hunk 1 above)
Comment (antiguru) [reply doc/user/content/sql/system-catalog.md:273]: Sure, we can do that. I named it in a way that aligned with the existing names, which come from librdkafka (not sure if they are also part of Kafka.)

**Comment by elindsey** [comment]
Is this for #7142 ?

I might be misunderstanding the intent, but I thought what we wanted is a repeatedly updated hi offset that we can compare the downstream view offsets against to generate a 'view lag from kafka tip' for measuring up-to-dateness instead of a single hi offset as seen at startup? We may already have that updating hi offset in a table, so I thought what's missing is the ts/offset mapping to do the join with view progress.

**Comment by antifuchs** [comment]
> I thought what we wanted is a repeatedly updated hi offset that we can compare the downstream view offsets against to generate a 'view lag from kafka tip' for measuring up-to-dateness instead of a single hi offset as seen at startup?

So that's explicitly not that. In #4990, my impression is that the question we wanted answered (and that we _can_ answer reasonably) is "has this view caught up to where it should have been at creation / startup time". That, AIUI, is a question that we can even answer (because the goalpost doesn't move), so I suggested we focus on this one for now. Is that not how it should go?

**Pull request merged**